frontend/src/modules/entitydetails/components/Metadata.js

```diff
@@ -0,0 +1,140 @@
+    render(): React.Node {
+        const { children, title } = this.props;
+        const className = title.trim().toLowerCase().replace(/ /g, '-');
+        return <p className={ className }>
+            <span className="title">{ title }</span>
+            <span className="content">{ children }</span>
+        </p>;
+    }
+}
+
+
+type Props = {|
+    entity: DbEntity,
+    locale: string,
+|};
+
+
+/**
+ * Component showing metadata of an entity.
+ *
+ * Shows:
+ *  - the original string
+ *  - a comment (if any)
+ *  - a context (if any)
+ *  - a list of source files (if any)
+ *  - a link to the resource
+ *  - a link to the project
+ */
+export default class Metadata extends React.Component<Props> {
+    renderComment(entity: DbEntity): React.Node {
+        if (!entity.comment) {
+            return null;
+        }
+
+        let comment = entity.comment;
+
+        const parts = entity.comment.split('\n');
+        if (parts[0].startsWith('MAX_LENGTH')) {
+            // This comment contains a max length instruction. Remove that part.
+            parts.shift();
+            comment = parts.join('\n');
+        }
+
+        return <Property title='Comment'>
+            <Linkify properties={ { target: '_blank', rel: 'noopener noreferrer' } }>
+                { comment }
+            </Linkify>
+        </Property>;
+    }
+
+    renderContext(entity: DbEntity): React.Node {
+        if (!entity.key) {
+            return null;
+        }
+
+        return <Property title='Context'>{ entity.key }</Property>;
+    }
+
+    renderSourceArray(source: Array<Array<string>>): React.Node {
+        return <ul>{ source.map((value, key) => {
+            return <li key={ key }><span className="title">#:</span>{ value.join(':') }</li>;
+        }) }</ul>;
+    }
+
+    renderSourceObject(source: Object): React.Node {
+        const examples = Object.keys(source).map(value => {
+            if (!source[value].example) {
+                return null;
+            }
+            return `$${value.toUpperCase()}$: ${source[value].example}`;
+        });
+
+        if (!examples) {
+            return null;
+        }
+
+        return <Property title='Placeholder Examples'>
+            <Linkify properties={ { target: '_blank', rel: 'noopener noreferrer' } }>
+                { examples.join(', ') }
+            </Linkify>
+        </Property>;
+    }
+
+    renderSources(entity: DbEntity): React.Node {
+        if (!entity.source) {
+            return null;
+        }
+
+        if (Array.isArray(entity.source)) {
+            return this.renderSourceArray(entity.source);
+        }
+
+        return this.renderSourceObject(entity.source);
+    }
+
+    render(): React.Node {
+        const { entity, locale } = this.props;
+
+        return <div className="metadata">
+            <Screenshots source={ entity.comment } locale={ locale } />
+            <p className="original">{ entity.original }</p>
```

Comment:
The original string is not part of string metadata. It should be a sibling of `div.metadata`, instead of a child.

Which means it should be a separate component. Although very simple right now, it will be much more complex in the case of FTL strings.

Reply:
I understand, but I think I'd rather refactor when the time comes to integrate support for Fluent. It's not harmful to not have a separate component right now, and when that time comes to have one, we might have a different idea for implementation.